3.analysis/scripts/nbconverted/2.gene_set_enrichment_analysis.py

```diff
@@ -75,30 +74,50 @@
     results.append(result)
     all_GSEA_results.append(result.assign(z_dim=f"z_{col}"))
     all_signatures.append(df)
-    neg_df = df.copy()
-    neg_df[col] = neg_df[col].sample(frac=1).reset_index(drop=True)
+
+
+# In[7]:
+
+
+# Generating negative control model for gsea
+
+neg_signature = signature.iloc[:, 1:].copy()
+
+for col in neg_signature.columns:
+    neg_signature.loc[:, col] = np.random.permutation(neg_signature.loc[:, col].values)
+
+genes = signature.iloc[:,:1]
+neg_signature.insert(0,'0', genes)
+
```

Comment:
Consider adding a comment here detailing how you are using the negative control and signatures in this for loop

Reply:
Good idea, adding that now